jina/peapods/pods/k8slib/kubernetes_tools.py

```diff
@@ -49,42 +51,52 @@ def k8s_client(self):
 
         :return: k8s client
         """
-        if self.__k8s_client is None:
-            self.__instantiate()
-        return self.__k8s_client
+        if self._k8s_client is None:
+            self._instantiate()
+        return self._k8s_client
 
     @property
-    def v1(self):
+    def core_v1(self):
         """V1 client for core
 
         :return: v1 client
         """
-        if self.__v1 is None:
-            self.__instantiate()
-        return self.__v1
+        if self._core_v1 is None:
+            self._instantiate()
+        return self._core_v1
+
+    @property
+    def apps_v1(self):
+        """V1 client for core
+
+        :return: v1 client
+        """
+        if self._apps_v1 is None:
+            self._instantiate()
+        return self._apps_v1
 
     @property
     def beta(self):
         """Beta client for using beta features
 
         :return: beta client
         """
-        if self.__beta is None:
-            self.__instantiate()
-        return self.__beta
+        if self._beta is None:
+            self._instantiate()
+        return self._beta
 
     @property
     def networking_v1_beta1_api(self):
         """Networking client used for creating the ingress
 
         :return: networking client
         """
-        if self.__networking_v1_beta1_api is None:
-            self.__instantiate()
-        return self.__networking_v1_beta1_api
+        if self._networking_v1_beta1_api is None:
+            self._instantiate()
+        return self._networking_v1_beta1_api
 
 
-__k8s_clients = K8SClients()
+_k8s_clients = K8sClients()
```

Comment:
shall we consider to make it public? I made a very first draft without creating template [here](https://github.com/jina-ai/jina/pull/3502/files#diff-763dd0d69c0c069bce44805ebe35b455d4c9bbea3819a122ca64333e514f2040R104) which takes a client as the first argument.

Reply:
We can consider later, I am not sure why it is so important to keep this as "singleton" style, should not create such big objects. I'd rather have every user instantiate one